browser-test/src/program_deep_link.test.ts

```diff
@@ -135,4 +135,21 @@ describe('navigating to a deep link', () => {
 
     await logout(page)
   })
+
+  it('Going to a deep link does not retain redirect in session', async () => {
+    await resetContext(ctx)
+    const {page, browserContext} = ctx
+    await browserContext.clearCookies()
+
+    // Go to a deep link
+    await gotoEndpoint(page, '/programs/test-deep-link')
+    await page.click('text="Continue to application"')
+
+    // Logging out should not take us back to /programs/test-deep-link, but rather
+    // to the program index page.
+    await logout(page)
+    expect(await page.innerText('h1')).toContain(
+      'Save time when applying for benefits',
+    )
+  })
```

Comment:
Confirmed the above fails before the changes to `RedirectController`, and passes after.